l10n_nl_bsn/models/res_partner.py

```diff
@@ -66,3 +67,30 @@ def _warn_bsn_existing(self):
             'message': msg % (self.name, self.bsn_number)
         }
         return warning
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        res_domain = []
+        for domain in args:
+            if (
+                    len(domain) > 2 and domain[0] == 'bsn_number' and
+                    isinstance(domain[2], str) and domain[2]
+            ):
+                operator = domain[1]
+                bsn_number = domain[2]
+                bsn_compact = bsn.compact(bsn_number)
+                bsn_domain = expression.OR([
+                    [('bsn_number', operator, bsn_number)],
+                    [('bsn_number', operator, bsn_compact)],
+                ])
+                if bsn.is_valid(bsn_number):
+                    bsn_format = bsn.format(bsn_number)
+                    bsn_domain = expression.OR([
+                        bsn_domain,
+                        [('bsn_number', operator, bsn_format)],
+                    ])
+                res_domain += bsn_domain
+            else:
+                res_domain.append(domain)
```

Comment:
Something like the below is probably a very bad idea for so many reasons, but at least will support that `ilike` search:

```
operator = domain[1]
try:
    bsn_number = domain[2].replace('.', '')
except AttributeError:
    bsn_number = domain[2]
sql = expression.expression([('bsn_number', operator, bsn_number)], self)).to_sql()
sql[0].replace('"res_partner"."bsn_number"::text', "replace(bsn_number, '.', '')")
self.env.cr.execute('select id from res_partner where ' + sql[0] % (sql[1],))
ids = self.env.cr.fetchall(...)
return super().search([('id', 'in', ids), offset=offset, limit=limit, order=order, count=count)])
```

Reply:
I got the idea :smirk: but my doubt is: does it worth it?

Reply:
Not now, unless a customer asks for it and keeps insisting and we cannot talk sense :-)

Let's place the limitation in README in "Known issues" and go with your improvement